src/os.rs

```diff
@@ -576,27 +576,60 @@ mod imp {
 
 #[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
 mod imp {
-    use std::io;
+    use stdweb::unstable::TryInto;
+    use stdweb::web::error::Error as WebError;
+    use {Error, ErrorKind};
 
     #[derive(Debug)]
     pub struct OsRng;
 
     impl OsRng {
         pub fn new() -> Result<OsRng, Error> {
-            Err(Error::new(ErrorKind::Unavailable, "not supported on WASM"))
+            Ok(OsRng)
```

Comment:
It would be nice if we could know here if `OsRng` is available or not. Would it add a lot of overhead to test here?

Reply:
We can check if `window.crypto.getRandomValues`、`require("crypto")` is available, but there is a certain overhead.

Reply:
I'm not really sure it matters — any of the other system interfaces may succeed with `new` but fail on `try_fill_bytes`.